Clean up errors sent to the client during Topic Unload

eolivelli opened the pull request:
### Motivation

While investigating on #1281 I have found a couple of enhancements:
- we should add more logging to troubleshoot unknown errors during topic unload
- we can improve some logging
- in case of CursorAlreadyClosedException  we return UNKNOWN_SERVER_ERROR and NOT_LEADER_FOR_PARTITION and this generates a "received an unknown error" log line on the client, that is pretty scary

### Modifications

- improve the toString representation of the callback passed to asyncFindPosition
- return NOT_LEADER_FOR_PARTITION in case of CursorAlreadyClosedException and ManagedLedgerFencedException
- reduce log level of some parts of the code

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

BewareMyPower commented:
Could you give the correct issue number? I see you still mentioned this PR after the change.

eolivelli commented:
> Could you give the correct issue number? I see you still mentioned this PR after the change.

sorry, it was #1281. I closed the issue, it was a configuration problem. Btw I added these useful little enhancements that helped me debugging the problem and also we removed some scary logs on the client

BewareMyPower commented on kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageMetadataUtils.java:
```diff
@@ -191,6 +204,12 @@ public static CompletableFuture<Position> asyncFindPosition(final ManagedLedger
+        }
+
+        @Override
+        public String toString() {
+            return "FindEntryByOffset{ " + offset + "}";
+        }
```
Good find! I think we can add the info of the managed ledger like

```java
            return "FindEntryByOffset{ managed ledger: " + managedLedger.getName() + ", offset: " + offset + " }";
```

And we can only reference the name when constructing the predicate.

```java
        return managedLedger.asyncFindPosition(new FindEntryByOffset(managedLedger.getName(),
                offset, skipMessagesWithoutIndex));
    }

    @AllArgsConstructor
    private static class FindEntryByOffset implements Predicate<Entry> {

        private final String managedLedgerName;
```

eolivelli replied:
IIRC the topic name is already reported in the logs of asyncFindPosition:

https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1710
